### ISSUE TITLE
WebApp:Set-AzWebApp is not honoring -AppSettings

### ISSUE THREAD

**Issue by elhutchi** [Azure/azure-powershell#14998]
## Description

Within the last few days, when calling `Set-AzWebApp` to only change app settings, the old values stay in place rather than being updated with the value provided for `-AppSettings`.

This is because if no `SiteConfig` properties are passed in ([see here](https://github.com/Azure/azure-powershell/blob/afcd29fee5c87458255f93d09398efff7550fd4d/src/Websites/Websites/Cmdlets/WebApps/SetAzureWebApp.cs#L165..L198)), the existing `SiteConfig` object is loaded and PATCHed after the App Settings are PUT ([see here](https://github.com/Azure/azure-powershell/blob/afcd29fee5c87458255f93d09398efff7550fd4d/src/Websites/Websites/Utilities/WebsitesClient.cs#L516..L532)).

## Steps to reproduce

This doesn't update the App Settings:

```powershell
PS ~>$webapp.SiteConfig.AppSettings

Name      Value
----      -----
MySetting OriginalValue

PS ~>$webapp = $webapp | Set-AzWebApp -ResourceGroupName $ResourceGroupName -AppSettings @{"MySetting"="UpdatedValue"}
PS ~>$webapp.SiteConfig.AppSettings

Name      Value
----      -----
MySetting OriginalValue
```

However this does update the App Settings, because we're setting another `SiteConfig` property (`-Use32BitWorkerProcess`).

```powershell
PS ~>$webapp = $webapp | Set-AzWebApp -ResourceGroupName $ResourceGroupName -AppSettings @{"MySetting"="UpdatedValue"} -Use32BitWorkerProcess $true
PS ~>$webapp.SiteConfig.AppSettings

Name      Value
----      -----
MySetting UpdatedValue
```

## Environment data

```
Name                           Value
----                           -----
PSVersion                      7.1.3
PSEdition                      Core
GitCommitId                    7.1.3
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

## Module versions

```powershell
ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Script     5.9.0                 Az                                  Core,Desk
Script     2.2.8                 Az.Accounts                         Core,Desk {Disable-AzDataCollection, Disable-AzContextAutosave, Enable-AzDataCollection, Enable-AzContextAutosave…}
Script     1.1.1                 Az.Advisor                          Core,Desk {Get-AzAdvisorRecommendation, Enable-AzAdvisorRecommendation, Disable-AzAdvisorRecommendation, Get-AzAdvisorConfiguration…}
Script     2.1.0                 Az.Aks                              Core,Desk {Get-AzAksCluster, New-AzAksCluster, Remove-AzAksCluster, Import-AzAksCredential…}
Script     1.1.4                 Az.AnalysisServices                 Core,Desk {Resume-AzAnalysisServicesServer, Suspend-AzAnalysisServicesServer, Get-AzAnalysisServicesServer, Remove-AzAnalysisServicesServe…
Script     2.2.0                 Az.ApiManagement                    Core,Desk {Add-AzApiManagementApiToGateway, Add-AzApiManagementApiToProduct, Add-AzApiManagementProductToGroup, Add-AzApiManagementRegion…}
Script     1.0.0                 Az.AppConfiguration                 Core,Desk {Get-AzAppConfigurationStore, Get-AzAppConfigurationStoreKey, New-AzAppConfigurationStore, New-AzAppConfigurationStoreKey…}
Script     1.1.0                 Az.ApplicationInsights              Core,Desk {Get-AzApplicationInsights, New-AzApplicationInsights, Remove-AzApplicationInsights, Update-AzApplicationInsights…}
Script     1.7.0                 Az.Automation                       Core,Desk {Get-AzAutomationHybridWorkerGroup, Remove-AzAutomationHybridWorkerGroup, Get-AzAutomationJobOutputRecord, Import-AzAutomationDs…
Script     3.1.0                 Az.Batch                            Core,Desk {Remove-AzBatchAccount, Get-AzBatchAccount, Get-AzBatchAccountKey, New-AzBatchAccount…}
Script     2.0.0                 Az.Billing                          Core,Desk {Get-AzBillingInvoice, Get-AzBillingPeriod, Get-AzEnrollmentAccount, Get-AzConsumptionBudget…}
Script     1.7.0                 Az.Cdn                              Core,Desk {Get-AzCdnProfile, Get-AzCdnProfileSsoUrl, New-AzCdnProfile, Remove-AzCdnProfile…}
Script     1.8.0                 Az.CognitiveServices                Core,Desk {Get-AzCognitiveServicesAccount, Get-AzCognitiveServicesAccountKey, Get-AzCognitiveServicesAccountSku, Get-AzCognitiveServicesAc…
Script     4.12.0                Az.Compute                          Core,Desk {Remove-AzAvailabilitySet, Get-AzAvailabilitySet, New-AzAvailabilitySet, Update-AzAvailabilitySet…}
Script     1.0.3                 Az.ContainerInstance                Core,Desk {New-AzContainerGroup, Get-AzContainerGroup, Remove-AzContainerGroup, Get-AzContainerInstanceLog}
Script     2.2.2                 Az.ContainerRegistry                Core,Desk {New-AzContainerRegistry, Get-AzContainerRegistry, Update-AzContainerRegistry, Remove-AzContainerRegistry…}
Script     1.1.0                 Az.CosmosDB                         Core,Desk {Get-AzCosmosDBSqlContainer, Get-AzCosmosDBSqlContainerThroughput, Get-AzCosmosDBSqlDatabase, Get-AzCosmosDBSqlDatabaseThroughpu…
Script     1.1.0                 Az.DataBoxEdge                      Core,Desk {Get-AzDataBoxEdgeJob, Get-AzDataBoxEdgeDevice, Invoke-AzDataBoxEdgeDevice, New-AzDataBoxEdgeDevice…}
Script     1.1.0                 Az.Databricks                       Core,Desk {Get-AzDatabricksVNetPeering, Get-AzDatabricksWorkspace, New-AzDatabricksVNetPeering, New-AzDatabricksWorkspace…}
Script     1.12.0                Az.DataFactory                      Core,Desk {Set-AzDataFactoryV2, Update-AzDataFactoryV2, Get-AzDataFactoryV2, Remove-AzDataFactoryV2…}
Script     1.0.2                 Az.DataLakeAnalytics                Core,Desk {Get-AzDataLakeAnalyticsDataSource, New-AzDataLakeAnalyticsCatalogCredential, Remove-AzDataLakeAnalyticsCatalogCredential, Set-A…
Script     1.3.0                 Az.DataLakeStore                    Core,Desk {Get-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreFirewallRule, Set-AzDataLa…
Script     1.0.0                 Az.DataShare                        Core,Desk {New-AzDataShareAccount, Get-AzDataShareAccount, Remove-AzDataShareAccount, New-AzDataShare…}
Script     1.1.0                 Az.DeploymentManager                Core,Desk {Get-AzDeploymentManagerArtifactSource, New-AzDeploymentManagerArtifactSource, Set-AzDeploymentManagerArtifactSource, Remove-AzD…
Script     2.1.1                 Az.DesktopVirtualization            Core,Desk {Disconnect-AzWvdUserSession, Expand-AzWvdMsixImage, Get-AzWvdApplication, Get-AzWvdApplicationGroup…}
Script     1.0.2                 Az.DevTestLabs                      Core,Desk {Get-AzDtlAllowedVMSizesPolicy, Get-AzDtlAutoShutdownPolicy, Get-AzDtlAutoStartPolicy, Get-AzDtlVMsPerLabPolicy…}
Script     1.1.2                 Az.Dns                              Core,Desk {Get-AzDnsRecordSet, New-AzDnsRecordConfig, Remove-AzDnsRecordSet, Set-AzDnsRecordSet…}
Script     1.3.0                 Az.EventGrid                        Core,Desk {New-AzEventGridTopic, Get-AzEventGridTopic, Set-AzEventGridTopic, New-AzEventGridTopicKey…}
Script     1.7.2                 Az.EventHub                         Core,Desk {New-AzEventHubNamespace, Get-AzEventHubNamespace, Set-AzEventHubNamespace, Remove-AzEventHubNamespace…}
Script     1.8.0                 Az.FrontDoor                        Core,Desk {New-AzFrontDoor, Get-AzFrontDoor, Set-AzFrontDoor, Remove-AzFrontDoor…}
Script     2.0.0                 Az.Functions                        Core,Desk {Get-AzFunctionApp, Get-AzFunctionAppAvailableLocation, Get-AzFunctionAppPlan, Get-AzFunctionAppSetting…}
Script     4.2.1                 Az.HDInsight                        Core,Desk {Get-AzHDInsightJob, New-AzHDInsightSqoopJobDefinition, Wait-AzHDInsightJob, New-AzHDInsightStreamingMapReduceJobDefinition…}
Script     1.3.1                 Az.HealthcareApis                   Core,Desk {New-AzHealthcareApisService, Remove-AzHealthcareApisService, Set-AzHealthcareApisService, Get-AzHealthcareApisService}
Script     2.7.3                 Az.IotHub                           Core,Desk {Add-AzIotHubKey, Get-AzIotHubEventHubConsumerGroup, Get-AzIotHubConnectionString, Get-AzIotHubJob…}
Script     3.4.3                 Az.KeyVault                         Core,Desk {Add-AzKeyVaultCertificate, Update-AzKeyVaultCertificate, Stop-AzKeyVaultCertificateOperation, Get-AzKeyVaultCertificateOperatio…
Script     1.0.1                 Az.Kusto                            Core,Desk {Add-AzKustoClusterLanguageExtension, Add-AzKustoDatabasePrincipal, Get-AzKustoAttachedDatabaseConfiguration, Get-AzKustoCluster…
Script     1.5.0                 Az.LogicApp                         Core,Desk {Get-AzIntegrationAccountAgreement, Get-AzIntegrationAccountAssembly, Get-AzIntegrationAccountBatchConfiguration, Get-AzIntegrat…
Script     1.1.3                 Az.MachineLearning                  Core,Desk {Move-AzMlCommitmentAssociation, Get-AzMlCommitmentAssociation, Get-AzMlCommitmentPlanUsageHistory, Remove-AzMlCommitmentPlan…}
Script     1.1.0                 Az.Maintenance                      Core,Desk {Get-AzApplyUpdate, Get-AzConfigurationAssignment, Get-AzMaintenanceConfiguration, Get-AzMaintenanceUpdate…}
Script     2.0.0                 Az.ManagedServices                  Core,Desk {Get-AzManagedServicesAssignment, New-AzManagedServicesAssignment, Remove-AzManagedServicesAssignment, Get-AzManagedServicesDefi…
Script     1.0.2                 Az.MarketplaceOrdering              Core,Desk {Get-AzMarketplaceTerms, Set-AzMarketplaceTerms}
Script     1.1.1                 Az.Media                            Core,Desk {Sync-AzMediaServiceStorageKey, Set-AzMediaServiceKey, Get-AzMediaServiceKey, Get-AzMediaServiceNameAvailability…}
Script     1.0.1                 Az.Migrate                          Core,Desk {Get-AzMigrateDiscoveredServer, Get-AzMigrateJob, Get-AzMigrateProject, Get-AzMigrateReplicationFabric…}
Script     2.5.0                 Az.Monitor                          Core,Desk {Get-AzMetricDefinition, Get-AzMetric, Remove-AzLogProfile, Get-AzLogProfile…}
Script     4.7.0                 Az.Network                          Core,Desk {Add-AzApplicationGatewayAuthenticationCertificate, Get-AzApplicationGatewayAuthenticationCertificate, New-AzApplicationGatewayA…
Script     1.1.1                 Az.NotificationHubs                 Core,Desk {Get-AzNotificationHub, Get-AzNotificationHubAuthorizationRule, Get-AzNotificationHubListKey, Get-AzNotificationHubPNSCredential…
Script     2.3.0                 Az.OperationalInsights              Core,Desk {New-AzOperationalInsightsAzureActivityLogDataSource, New-AzOperationalInsightsCustomLogDataSource, Disable-AzOperationalInsight…
Script     1.4.1                 Az.PolicyInsights                   Core,Desk {Get-AzPolicyEvent, Get-AzPolicyState, Get-AzPolicyStateSummary, Get-AzPolicyRemediation…}
Script     1.1.2                 Az.PowerBIEmbedded                  Core,Desk {Remove-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollectionAccessKey, Get-AzPowerBI…
Script     1.0.3                 Az.PrivateDns                       Core,Desk {Get-AzPrivateDnsZone, Remove-AzPrivateDnsZone, Set-AzPrivateDnsZone, New-AzPrivateDnsZone…}
Script     3.6.0                 Az.RecoveryServices                 Core,Desk {Get-AzRecoveryServicesBackupProperty, Get-AzRecoveryServicesVault, Get-AzRecoveryServicesVaultSettingsFile, New-AzRecoveryServi…
Script     1.4.0                 Az.RedisCache                       Core,Desk {Remove-AzRedisCachePatchSchedule, New-AzRedisCacheScheduleEntry, Get-AzRedisCachePatchSchedule, New-AzRedisCachePatchSchedule…}
Script     1.0.0                 Az.RedisEnterpriseCache             Core,Desk {Export-AzRedisEnterpriseCache, Get-AzRedisEnterpriseCache, Get-AzRedisEnterpriseCacheDatabase, Get-AzRedisEnterpriseCacheKey…}
Script     1.0.3                 Az.Relay                            Core,Desk {New-AzRelayNamespace, Get-AzRelayNamespace, Set-AzRelayNamespace, Remove-AzRelayNamespace…}
Script     1.0.0                 Az.ResourceMover                    Core,Desk {Add-AzResourceMoverMoveResource, Get-AzResourceMoverMoveCollection, Get-AzResourceMoverMoveResource, Get-AzResourceMoverRequire…
Script     3.5.0                 Az.Resources                        Core,Desk {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzRoleAssignment, New-AzRoleAssignment…}
Script     1.5.0                 Az.ServiceBus                       Core,Desk {New-AzServiceBusNamespace, Get-AzServiceBusNamespace, Set-AzServiceBusNamespace, Remove-AzServiceBusNamespace…}
Script     2.4.0                 Az.ServiceFabric                    Core,Desk {Add-AzServiceFabricClientCertificate, Add-AzServiceFabricClusterCertificate, Add-AzServiceFabricNode, Add-AzServiceFabricNodeTy…
Script     1.2.0                 Az.SignalR                          Core,Desk {New-AzSignalR, Get-AzSignalR, Get-AzSignalRKey, New-AzSignalRKey…}
Script     2.17.1                Az.Sql                              Core,Desk {Get-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseTransparentDataEncryptionActivity, Set-AzSqlDatabaseTransparentDat…
Script     1.1.0                 Az.SqlVirtualMachine                Core,Desk {New-AzSqlVM, Get-AzSqlVM, Update-AzSqlVM, Remove-AzSqlVM…}
Script     3.6.0                 Az.Storage                          Core,Desk {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStorageAccount, New-AzStorageAccountKey…}
Script     1.4.0                 Az.StorageSync                      Core,Desk {Invoke-AzStorageSyncCompatibilityCheck, New-AzStorageSyncService, Get-AzStorageSyncService, Set-AzStorageSyncService…}
Script     1.1.0                 Az.StreamAnalytics                  Core,Desk {Get-AzStreamAnalyticsFunction, Get-AzStreamAnalyticsDefaultFunctionDefinition, New-AzStreamAnalyticsFunction, Remove-AzStreamAn…
Script     1.0.0                 Az.Support                          Core,Desk {Get-AzSupportService, Get-AzSupportProblemClassification, Get-AzSupportTicket, Get-AzSupportTicketCommunication…}
Script     1.0.4                 Az.TrafficManager                   Core,Desk {Add-AzTrafficManagerCustomHeaderToEndpoint, Remove-AzTrafficManagerCustomHeaderFromEndpoint, Add-AzTrafficManagerCustomHeaderTo…
Script     2.5.0                 Az.Websites                         Core,Desk {Get-AzAppServicePlan, Set-AzAppServicePlan, New-AzAppServicePlan, Remove-AzAppServicePlan…}
Binary     2.0.2.130             AzureAD                             Desk
Script     4.10.1                Pester                              Desk      {Describe, Context, It, Should…}
```

## Debug output

If you run the non-working and working examples above with `-Debug`, then diff the results, you can see how the failing example is caused by the PUT of the new app settings then PATCH of the old ones. 

Here is a short snippet of the non-working call:

```
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG-AppSettingsTest/providers/Microsoft.Web/sites/myappsettingstest/config/appsettings?api-version=2020-06-01

Headers:
...

Body:
{
  "properties": {
    "MySetting": "UpdatedValue"
  }
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
...

Body:
{
  "id": "/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG-AppSettingsTest/providers/Microsoft.Web/sites/myappsettingstest/config/appsettings",
  "name": "appsettings",
  "type": "Microsoft.Web/sites/config",
  "location": "North Central US",
  "properties": {
    "MySetting": "UpdatedValue"
  }
}


DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PATCH

Absolute Uri:
https://management.azure.com/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG-AppSettingsTest/providers/Microsoft.Web/sites/myappsettingstest/config/web?api-version=2020-06-01

Headers:
...

Body:
{
  "properties": {
    "numberOfWorkers": 1,
    "defaultDocuments": [
      "Default.htm",
      "Default.html",
      "Default.asp",
      "index.htm",
      "index.html",
      "iisstart.htm",
      "default.aspx",
      "index.php",
      "hostingstart.html"
    ],
    "netFrameworkVersion": "v4.0",
    "phpVersion": "5.6",
    "pythonVersion": "",
    "nodeVersion": "",
    "linuxFxVersion": "",
    "requestTracingEnabled": false,
    "remoteDebuggingEnabled": false,
    "remoteDebuggingVersion": "VS2019",
    "httpLoggingEnabled": false,
    "logsDirectorySizeLimit": 35,
    "detailedErrorLoggingEnabled": false,
    "publishingUsername": "$myappsettingstest",
    "appSettings": [
      {
        "name": "MySetting",
        "value": "OriginalValue"
      }
    ],
    "connectionStrings": [],
    "scmType": "None",
    "use32BitWorkerProcess": true,
    "webSocketsEnabled": false,
    "alwaysOn": false,
    "appCommandLine": "",
    "managedPipelineMode": "Integrated",
    "virtualApplications": [
      {
        "virtualPath": "/",
        "physicalPath": "site\\wwwroot",
        "preloadEnabled": false
      }
    ],
    "loadBalancing": "LeastRequests",
    "experiments": {
      "rampUpRules": []
    },
    "autoHealEnabled": false,
    "vnetName": "",
    "localMySqlEnabled": false,
    "ipSecurityRestrictions": [
      {
        "ipAddress": "Any",
        "action": "Allow",
        "priority": 1,
        "name": "Allow all",
        "description": "Allow all access"
      }
    ],
    "scmIpSecurityRestrictions": [
      {
        "ipAddress": "Any",
        "action": "Allow",
        "priority": 1,
        "name": "Allow all",
        "description": "Allow all access"
      }
    ],
    "scmIpSecurityRestrictionsUseMain": false,
    "http20Enabled": false,
    "minTlsVersion": "1.2",
    "ftpsState": "AllAllowed"
  }
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
...

Body:
{
  "id": "/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG-AppSettingsTest/providers/Microsoft.Web/sites/myappsettingstest",
  "name": "myappsettingstest",
  "type": "Microsoft.Web/sites",
  "location": "North Central US",
  "properties": {
    "numberOfWorkers": 1,
    "defaultDocuments": [
      "Default.htm",
      "Default.html",
      "Default.asp",
      "index.htm",
      "index.html",
      "iisstart.htm",
      "default.aspx",
      "index.php",
      "hostingstart.html"
    ],
    "netFrameworkVersion": "v4.0",
    "phpVersion": "5.6",
    "pythonVersion": "",
    "nodeVersion": "",
    "powerShellVersion": "",
    "linuxFxVersion": "",
    "windowsFxVersion": null,
    "requestTracingEnabled": false,
    "remoteDebuggingEnabled": false,
    "remoteDebuggingVersion": "VS2019",
    "httpLoggingEnabled": false,
    "azureMonitorLogCategories": null,
    "acrUseManagedIdentityCreds": false,
    "acrUserManagedIdentityID": null,
    "logsDirectorySizeLimit": 35,
    "detailedErrorLoggingEnabled": false,
    "publishingUsername": "$myappsettingstest",
    "publishingPassword": null,
    "appSettings": null,
    "azureStorageAccounts": {},
    "metadata": null,
    "connectionStrings": null,
    "machineKey": null,
    "handlerMappings": null,
    "documentRoot": null,
    "scmType": "None",
    "use32BitWorkerProcess": true,
    "webSocketsEnabled": false,
    "alwaysOn": false,
    "javaVersion": null,
    "javaContainer": null,
    "javaContainerVersion": null,
    "appCommandLine": "",
    "managedPipelineMode": "Integrated",
    "virtualApplications": [
      {
        "virtualPath": "/",
        "physicalPath": "site\\wwwroot",
        "preloadEnabled": false,
        "virtualDirectories": null
      }
    ],
    "winAuthAdminState": 0,
    "winAuthTenantState": 0,
    "customAppPoolIdentityAdminState": false,
    "customAppPoolIdentityTenantState": false,
    "runtimeADUser": null,
    "runtimeADUserPassword": null,
    "loadBalancing": "LeastRequests",
    "routingRules": [],
    "experiments": {
      "rampUpRules": []
    },
    "limits": null,
    "autoHealEnabled": false,
    "autoHealRules": null,
    "tracingOptions": null,
    "vnetName": "",
    "vnetRouteAllEnabled": false,
    "vnetPrivatePortsCount": 0,
    "publicNetworkAccess": null,
    "siteAuthEnabled": false,
    "siteAuthSettings": {
      "enabled": null,
      "configVersion": null,
      "unauthenticatedClientAction": null,
      "tokenStoreEnabled": null,
      "allowedExternalRedirectUrls": null,
      "defaultProvider": null,
      "clientId": null,
      "clientSecret": null,
      "clientSecretSettingName": null,
      "clientSecretCertificateThumbprint": null,
      "issuer": null,
      "allowedAudiences": null,
      "additionalLoginParams": null,
      "isAadAutoProvisioned": false,
      "aadClaimsAuthorization": null,
      "googleClientId": null,
      "googleClientSecret": null,
      "googleClientSecretSettingName": null,
      "googleOAuthScopes": null,
      "facebookAppId": null,
      "facebookAppSecret": null,
      "facebookAppSecretSettingName": null,
      "facebookOAuthScopes": null,
      "gitHubClientId": null,
      "gitHubClientSecret": null,
      "gitHubClientSecretSettingName": null,
      "gitHubOAuthScopes": null,
      "twitterConsumerKey": null,
      "twitterConsumerSecret": null,
      "twitterConsumerSecretSettingName": null,
      "microsoftAccountClientId": null,
      "microsoftAccountClientSecret": null,
      "microsoftAccountClientSecretSettingName": null,
      "microsoftAccountOAuthScopes": null
    },
    "cors": null,
    "push": null,
    "apiDefinition": null,
    "apiManagementConfig": null,
    "autoSwapSlotName": null,
    "localMySqlEnabled": false,
    "managedServiceIdentityId": null,
    "xManagedServiceIdentityId": null,
    "keyVaultReferenceIdentity": null,
    "ipSecurityRestrictions": [
      {
        "ipAddress": "Any",
        "action": "Allow",
        "priority": 1,
        "name": "Allow all",
        "description": "Allow all access"
      }
    ],
    "scmIpSecurityRestrictions": [
      {
        "ipAddress": "Any",
        "action": "Allow",
        "priority": 1,
        "name": "Allow all",
        "description": "Allow all access"
      }
    ],
    "scmIpSecurityRestrictionsUseMain": false,
    "http20Enabled": false,
    "minTlsVersion": "1.2",
    "scmMinTlsVersion": "1.0",
    "ftpsState": "AllAllowed",
    "preWarmedInstanceCount": 0,
    "functionAppScaleLimit": 0,
    "healthCheckPath": null,
    "fileChangeAuditEnabled": false,
    "functionsRuntimeScaleMonitoringEnabled": false,
    "websiteTimeZone": null,
    "minimumElasticInstanceCount": 0
  }
}

```

---

And here is the same section of the working call. Notice that the PATCH body is much shorter and doesn't contain the old app settings this time. 

And if you compare the diff yourself, you'll see only the new app settings are used beyond this point in the logs, while the non-working logs will only show the old app settings.

```
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG-AppSettingsTest/providers/Microsoft.Web/sites/myappsettingstest/config/appsettings?api-version=2020-06-01

Headers:
...

Body:
{
  "properties": {
    "MySetting": "UpdatedValue"
  }
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
...

Body:
{
  "id": "/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG-AppSettingsTest/providers/Microsoft.Web/sites/myappsettingstest/config/appsettings",
  "name": "appsettings",
  "type": "Microsoft.Web/sites/config",
  "location": "North Central US",
  "properties": {
    "MySetting": "UpdatedValue"
  }
}


DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PATCH

Absolute Uri:
https://management.azure.com/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG-AppSettingsTest/providers/Microsoft.Web/sites/myappsettingstest/config/web?api-version=2020-06-01

Headers:
...

Body:
{
  "properties": {
    "numberOfWorkers": 1,
    "use32BitWorkerProcess": true,
    "minTlsVersion": "1.2",
    "ftpsState": "AllAllowed"
  }
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
...

Body:
{
  "id": "/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG-AppSettingsTest/providers/Microsoft.Web/sites/myappsettingstest",
  "name": "myappsettingstest",
  "type": "Microsoft.Web/sites",
  "location": "North Central US",
  "properties": {
    "numberOfWorkers": 1,
    "defaultDocuments": [
      "Default.htm",
      "Default.html",
      "Default.asp",
      "index.htm",
      "index.html",
      "iisstart.htm",
      "default.aspx",
      "index.php",
      "hostingstart.html"
    ],
    "netFrameworkVersion": "v4.0",
    "phpVersion": "5.6",
    "pythonVersion": "",
    "nodeVersion": "",
    "powerShellVersion": "",
    "linuxFxVersion": "",
    "windowsFxVersion": null,
    "requestTracingEnabled": false,
    "remoteDebuggingEnabled": false,
    "remoteDebuggingVersion": "VS2019",
    "httpLoggingEnabled": false,
    "azureMonitorLogCategories": null,
    "acrUseManagedIdentityCreds": false,
    "acrUserManagedIdentityID": null,
    "logsDirectorySizeLimit": 35,
    "detailedErrorLoggingEnabled": false,
    "publishingUsername": "$myappsettingstest",
    "publishingPassword": null,
    "appSettings": null,
    "azureStorageAccounts": {},
    "metadata": null,
    "connectionStrings": null,
    "machineKey": null,
    "handlerMappings": null,
    "documentRoot": null,
    "scmType": "None",
    "use32BitWorkerProcess": true,
    "webSocketsEnabled": false,
    "alwaysOn": false,
    "javaVersion": null,
    "javaContainer": null,
    "javaContainerVersion": null,
    "appCommandLine": "",
    "managedPipelineMode": "Integrated",
    "virtualApplications": [
      {
        "virtualPath": "/",
        "physicalPath": "site\\wwwroot",
        "preloadEnabled": false,
        "virtualDirectories": null
      }
    ],
    "winAuthAdminState": 0,
    "winAuthTenantState": 0,
    "customAppPoolIdentityAdminState": false,
    "customAppPoolIdentityTenantState": false,
    "runtimeADUser": null,
    "runtimeADUserPassword": null,
    "loadBalancing": "LeastRequests",
    "routingRules": [],
    "experiments": {
      "rampUpRules": []
    },
    "limits": null,
    "autoHealEnabled": false,
    "autoHealRules": null,
    "tracingOptions": null,
    "vnetName": "",
    "vnetRouteAllEnabled": false,
    "vnetPrivatePortsCount": 0,
    "publicNetworkAccess": null,
    "siteAuthEnabled": false,
    "siteAuthSettings": {
      "enabled": null,
      "configVersion": null,
      "unauthenticatedClientAction": null,
      "tokenStoreEnabled": null,
      "allowedExternalRedirectUrls": null,
      "defaultProvider": null,
      "clientId": null,
      "clientSecret": null,
      "clientSecretSettingName": null,
      "clientSecretCertificateThumbprint": null,
      "issuer": null,
      "allowedAudiences": null,
      "additionalLoginParams": null,
      "isAadAutoProvisioned": false,
      "aadClaimsAuthorization": null,
      "googleClientId": null,
      "googleClientSecret": null,
      "googleClientSecretSettingName": null,
      "googleOAuthScopes": null,
      "facebookAppId": null,
      "facebookAppSecret": null,
      "facebookAppSecretSettingName": null,
      "facebookOAuthScopes": null,
      "gitHubClientId": null,
      "gitHubClientSecret": null,
      "gitHubClientSecretSettingName": null,
      "gitHubOAuthScopes": null,
      "twitterConsumerKey": null,
      "twitterConsumerSecret": null,
      "twitterConsumerSecretSettingName": null,
      "microsoftAccountClientId": null,
      "microsoftAccountClientSecret": null,
      "microsoftAccountClientSecretSettingName": null,
      "microsoftAccountOAuthScopes": null
    },
    "cors": null,
    "push": null,
    "apiDefinition": null,
    "apiManagementConfig": null,
    "autoSwapSlotName": null,
    "localMySqlEnabled": false,
    "managedServiceIdentityId": null,
    "xManagedServiceIdentityId": null,
    "keyVaultReferenceIdentity": null,
    "ipSecurityRestrictions": [
      {
        "ipAddress": "Any",
        "action": "Allow",
        "priority": 1,
        "name": "Allow all",
        "description": "Allow all access"
      }
    ],
    "scmIpSecurityRestrictions": [
      {
        "ipAddress": "Any",
        "action": "Allow",
        "priority": 1,
        "name": "Allow all",
        "description": "Allow all access"
      }
    ],
    "scmIpSecurityRestrictionsUseMain": false,
    "http20Enabled": false,
    "minTlsVersion": "1.2",
    "scmMinTlsVersion": "1.0",
    "ftpsState": "AllAllowed",
    "preWarmedInstanceCount": 0,
    "functionAppScaleLimit": 0,
    "healthCheckPath": null,
    "fileChangeAuditEnabled": false,
    "functionsRuntimeScaleMonitoringEnabled": false,
    "websiteTimeZone": null,
    "minimumElasticInstanceCount": 0
  }
}
```

## Error output

N/A

**Comment by Kotasudhakarreddy** [comment]
@elhutchi Thanks for your feedback! i am able to repro this issue. We will fix this in the coming releases.

**Comment by TheoroiMS** [comment]
Hi @Kotasudhakarreddy . Not sure if you are aware of. The same issue also happen for Set-AzWebAppSlot.

**Comment by MaxChal** [comment]
Hi @Kotasudhakarreddy, we are having a similar issue. We are using Set-AzWebAppSlot and the config temporarily updates whilst the command is running, but then all changes disappears once the command is complete. 

No errors in the log and the Powershell command completes successfully.

**Comment by Kotasudhakarreddy** [comment]
Yes `Set-AzWebAppSlot` also has been take care in the given fix.

**Comment by leshy84** [comment]
Hello, we also bumped into the problem a couple of days ago, but I do not remember that we updated the powershell modules last weeks.
Is anything changed in Azure the last week?

**Comment by panchagnula** [comment]
> Yes `Set-AzWebAppSlot` also has been take care in the given fix.

@Kotasudhakarreddy is this a regression? Do we know what regressed this? Do we have a test to catch such regressions in future?

**Comment by Kotasudhakarreddy** [comment]
> > Yes `Set-AzWebAppSlot` also has been take care in the given fix.
> 
> @Kotasudhakarreddy is this a regression? Do we know what regressed this? Do we have a test to catch such regressions in future?
Yes it is a regression from this [fix](https://github.com/Azure/azure-powershell/pull/12107/files#diff-f5081b48a4605e5dc62abd9ed7239d86e85f6f9abf20a420adcfb979bde0a472L542). Checking with PR owner on the same.

**Comment by vivere-dally** [comment]
I can also reproduce this. I was wondering why it isn't working. Nothing has changed on my side, I guess the Cloud Api got updated or something. Please look into this since all deployments are failing with the lack of this feature. I have to manually add the app settings to have a working environment.

**Comment by LukeDearden** [comment]
Set-AzWebApp impacting us here. New deployments being created without AppSettings. 

As a workaround I found PowerShell 5 and Set-AzureRmWebApp still works.

Is there any ETA on the next release?

**Comment by LukeDearden** [comment]
Also as per https://github.com/Azure/azure-powershell/issues/15038#issuecomment-843612008 you can work around by adding -HttpLoggingEnabled $true

**Comment by AWahlqvist** [comment]
Also eagerly awaiting a fix for this.

We also discovered another issue with this cmdlet today though, it seems to unintentionally disable AlwaysOn even if that parameter is not specified. For example:
Set-AzWebAppSlot -FtpsState FtpsOnly -ResourceGroupName <my-rg> -Name <-my-web> -Slot <slot>

Will set AlwaysOn to Off when ran on a few of our subscriptions. It doesn't seem to be completely consistent but I can reproduce the error easily on the affected web apps.

Will a fix for that be part of this one or should I open a separate issue for it?

**Comment by panchagnula** [comment]
> Also eagerly awaiting a fix for this.
> 
> We also discovered another issue with this cmdlet today though, it seems to unintentionally disable AlwaysOn even if that parameter is not specified. For example:
> Set-AzWebAppSlot -FtpsState FtpsOnly -ResourceGroupName my-rg -Name my-web -Slot slot
> 
> Will set AlwaysOn to Off when ran on a few of our subscriptions. It doesn't seem to be completely consistent but I can reproduce the error easily on the affected web apps.
> 
> Will a fix for that be part of this one or should I open a separate issue for it?

Yes please file a separate issue so we can track it.

**Comment by AWahlqvist** [comment]
Done, see issue #15066. Thanks!

**Comment by panchagnula** [comment]
> Can we get any clarification as to why this got broken a few days ago without any change on the client (Azure Powershell / Az.Website) side, has there been a change to the validation or processing logic on the server (Azure Webapp API) side?
> 
> If yes, this is a serious problem going beyond the site config update logic: Either the client (Azure Powershell) does not properly specify the API version when making the API call, or the server (Azure Webapp API) has improperly changed the behavior of a versioned API call. Glancing at the Debug logs in OP, it looks like both API calls are specifying `?api-version=2020-06-01` so that suggests the latter 🤯

This was a regression from a PS command change - https://github.com/Azure/azure-powershell/pull/12107 the reason we believe most customers are setting this now because new PowerShell version with this fix is now being used. This was a definite miss on our end & we have ensured to add better testing on our end to not repeat the. Hope this helps.

**Comment by lerlacher-haylix** [comment]
Hi @panchagnula, thank you for your response. How can I identify whether I am using an affected version? I am using powershell on Ubuntu 18.04 and I have not recently updated powershell or any modules.

Here are the module versions I am using:

```
PS /> Get-Module                                                                                                                                                                                                                                                                                                                                                                                                                                        ModuleType Version    PreRelease Name                                ExportedCommands                                                                                                                                                                                          ---------- -------    ---------- ----                                ----------------
Script     1.9.4                 Az.Accounts                         {Add-AzEnvironment, Clear-AzContext, Clear-AzDefault, Connect-AzAccount…}
Script     1.11.0                Az.Websites                         {Add-AzWebAppAccessRestrictionRule, Add-AzWebAppTrafficRouting, Edit-AzWebAppBackupConfiguration, Enter-AzWebAppContainerPSSession…}
Manifest   7.0.0.0               Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty…}
Manifest   7.0.0.0               Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object…}
Script     2.1.0                 PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler…}
```

This is the powershell version I am using:

```
root@ubu:~# apt show powershell
Package: powershell
Version: 7.1.3-1.ubuntu.18.04
Priority: extra
Section: shells
Maintainer: PowerShell Team <PowerShellTeam@hotmail.com>
Installed-Size: 178 MB
Depends: libc6, libgcc1, libgssapi-krb5-2, liblttng-ust0, libstdc++6, zlib1g, libicu72|libicu71|libicu70|libicu69|libicu68|libicu67|libicu66|libicu65|libicu63|libicu60|libicu57|libicu55|libicu52, libssl1.1|libssl1.0.2|libssl1.0.0
Homepage: https://microsoft.com/powershell
License: MIT License
Vendor: Microsoft Corporation
Download-Size: 68.3 MB
APT-Manual-Installed: yes
APT-Sources: https://packages.microsoft.com/ubuntu/18.04/prod bionic/main amd64 Packages
Description: PowerShell is an automation and configuration management platform.
 It consists of a cross-platform command-line shell and associated scripting language.
```

**Comment by panchagnula** [comment]
> Hi @panchagnula, thank you for your response. How can I identify whether I am using an affected version? I am using powershell on Ubuntu 18.04 and I have not recently updated powershell or any modules.
> 
> Here are the module versions I am using:
> 
> ```
> PS /> Get-Module                                                                                                                                                                                                                                                                                                                                                                                                                                        ModuleType Version    PreRelease Name                                ExportedCommands                                                                                                                                                                                          ---------- -------    ---------- ----                                ----------------
> Script     1.9.4                 Az.Accounts                         {Add-AzEnvironment, Clear-AzContext, Clear-AzDefault, Connect-AzAccount…}
> Script     1.11.0                Az.Websites                         {Add-AzWebAppAccessRestrictionRule, Add-AzWebAppTrafficRouting, Edit-AzWebAppBackupConfiguration, Enter-AzWebAppContainerPSSession…}
> Manifest   7.0.0.0               Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty…}
> Manifest   7.0.0.0               Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object…}
> Script     2.1.0                 PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler…}
> 
> PS /> Get-Module Az
> 
> ModuleType Version    PreRelease Name                                ExportedCommands
> ---------- -------    ---------- ----                                ----------------
> Script     4.7.0                 Az
> ```
> 
> This is the powershell version I am using:
> 
> ```
> root@ubu:~# apt show powershell
> Package: powershell
> Version: 7.1.3-1.ubuntu.18.04
> Priority: extra
> Section: shells
> Maintainer: PowerShell Team <PowerShellTeam@hotmail.com>
> Installed-Size: 178 MB
> Depends: libc6, libgcc1, libgssapi-krb5-2, liblttng-ust0, libstdc++6, zlib1g, libicu72|libicu71|libicu70|libicu69|libicu68|libicu67|libicu66|libicu65|libicu63|libicu60|libicu57|libicu55|libicu52, libssl1.1|libssl1.0.2|libssl1.0.0
> Homepage: https://microsoft.com/powershell
> License: MIT License
> Vendor: Microsoft Corporation
> Download-Size: 68.3 MB
> APT-Manual-Installed: yes
> APT-Sources: https://packages.microsoft.com/ubuntu/18.04/prod bionic/main amd64 Packages
> Description: PowerShell is an automation and configuration management platform.
>  It consists of a cross-platform command-line shell and associated scripting language.
> ```

Looking at release history the regression was introduced in this release & existed since https://github.com/Azure/azure-powershell/releases/tag/v4.3.0-June2020

**Comment by lerlacher-haylix** [comment]
So that was released in June 2020 - how does that fit with the issues popping up this week?

**Comment by panchagnula** [comment]
> So that was released in June 2020 - how does that fit with the issues popping up this week?

Like I mentioned above the issue was introduced & existed since. So any build being used since then to now; when using only app settings in the command without other SiteProperties in the same command would be setting this.

**Comment by leshy84** [comment]
> > So that was released in June 2020 - how does that fit with the issues popping up this week?
> 
> Like I mentioned above the issue was introduced & existed since. So any build being used since then to now; when using only app settings in the command without other SiteProperties in the same command would be setting this.

We mean we used the same version of Az.WebSites module and the same deployment script (specifying only app settings in the function) for months, but the problem appeared last week only, though nothing changed on our side (neither deployment script, nor powershell modules)

**Comment by sadomovalex** [comment]
from our side I may also confirm that we used Az 5.6.0 for 2 months (switched from AzureRM) in the same script and the following command was working as expected i.e. app settings were saved properly:
```powershell
Set-AzWebApp -Name "MyAppService" -ResourceGroupName "MyResourceGroup" -AppSettings $appSettings
```
and it stopped working only on the last week. So it doesn't look like this issue is caused by change made about year ago.

**Comment by Kotasudhakarreddy** [comment]
Hi Fix is available in the latest `Az.websites` module. You can update your modules to latest one using `update-module -Name "Az.Websites"`. Please check and let us know the status. thanks!

**Comment by elhutchi** [comment]
This fixed it for me. I installed `Az` 6.0.0 which includes `Az.Websites` 2.6.0.

However as noted by others, previous versions of `Az.Websites` still have this newly introduced bug. Seems to me that maybe there was an underlying API change (fix?) that surfaced this bug in `Az.Websites`. Is it possible to fix this for older versions of the Az module too (such as 5.7.0 which is currently the latest available on Microsoft hosted pipeline agents)? If so, that should help prevent existing deployments from breaking.

**Comment by panchagnula** [comment]
Closing this since the deployment of the fix is complete.
